### PR TITLE
Add delegation tests

### LINKED ISSUE #367: spl-token-cli: Add support for account delegates
The account delegate feature of SPL Token is not exposed in the `spl-token` command-line tool yet.

### PR DESCRIPTION
**Pull request by jmalcic**
This PR resolves #367 by adding tests for transferring and burning tokens when delegated to another account.

### CODE REVIEW

File: token/cli/src/main.rs [hunk 1]
```diff
@@ -3714,4 +3715,158 @@ mod tests {
...
+            &config,
+            &payer,
+            &[
+                "spl-token",
+                CommandName::Transfer.into(),
+                &token.to_string(),
+                "10",
+                &destination.to_string(),
+                "--fund-recipient",
+                "--allow-unfunded-recipient",
```
Comment (joncinque) [inline token/cli/src/main.rs:3780]: nit: Since you already created the `destination` on line 3726, you shouldn't need these flags

**Review by joncinque — COMMENTED** [review]
Looks great! Just one little micro-nit, then this is good to go

**Comment by joncinque** [comment]
Side note: that random test failure on token-swap was pretty interesting! #3334

**Comment by jmalcic** [comment]
> Looks like this just needs a `cargo fmt` to pass CI. I'll investigate and fix the token-swap error, sorry about that.

Whoops yes I've now adjusted my IDE settings so this shouldn't happen again 😅

**Comment by joncinque** [comment]
Closing / opening to redo all CI

**Pull request merged**